feat: Kl/interop support

kelemeno opened the pull request:
## What ❔

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What are the principles behind them? -->
<!-- Example: PR templates ensure PR reviewers, observers, and future iterators are in context about the evolution of repos. -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.

kelemeno commented on core/tests/ts-integration/tests/interop.test.ts:
```diff
@@ -0,0 +1,217 @@
+    'L2AssetRouter'
+).abi;
+
+const ntvInterface = readContract(
```
@Raid5594

kelemeno commented on core/tests/ts-integration/package.json:
```diff
@@ -32,6 +32,7 @@
     "ts-node": "^10.1.0",
     "typescript": "^4.3.5",
     "zksync-ethers": "^6.9.0",
+    "zksync-ethers-interop-support": "git+https://github.com/zksync-sdk/zksync-ethers#kl/interop-ethers-support",
```
@Raid5594

nikitastupin-matterlabs commented on docs/src/specs/contracts/bridging/interop/overview.md:
```diff
@@ -0,0 +1,50 @@
+
+   - The user's EOA calls the InteropCenter contract. In the standard asset bridging scenario, they will call `requestL2TransactionSingleCall`. See the [InteropCenter](./interop_center/overview.md) documentation for additional options.
+   - The InteropCenter emits two `InteropBundleSent` events and a single `InteropTriggerSent` event. Together, these events specify the interop transaction. One bundle is used to pay for gas fees, while the other is the main execution bundle used to bridge the asset. The trigger specifies the interop transaction parameters so that the transaction can be executed automatically on the destination c ...
+   - For each event, an L2→L1 message is created that contains the event data. Each L2→L1 message is included in the chain's Merkle tree of emitted transactions in the `L2toL1Logs` Merkle tree (see [here](../../settlement_contracts/priority_queue/l1_l2_communication/l2_to_l1.md)). This inclusion makes the messages verifiable using Merkle proofs.
```
What is `chain's Merkle tree of emitted transactions`? What is `the L2toL1Logs Merkle tree`? How are they related? The reference doesn't define these terms.

nikitastupin-matterlabs commented on docs/src/specs/contracts/bridging/interop/overview.md:
```diff
@@ -0,0 +1,50 @@
+   - For each event, an L2→L1 message is created that contains the event data. Each L2→L1 message is included in the chain's Merkle tree of emitted transactions in the `L2toL1Logs` Merkle tree (see [here](../../settlement_contracts/priority_queue/l1_l2_communication/l2_to_l1.md)). This inclusion makes the messages verifiable using Merkle proofs.
+
+2. **Settlement of the Sending Chain on the Settlement Layer (SL)**
+   - The sending chain submits its proof to its settlement layer (L1 or Gateway). See [here](../../settlement_contracts/zkchain_basics.md) for more details.
```
What contract and which function are called? The `ExecutorFacet::proveBatchesSharedBridge` if I remember correctly?

nikitastupin-matterlabs commented on docs/src/specs/contracts/bridging/interop/overview.md:
```diff
@@ -0,0 +1,50 @@
+   - The sending chain submits its proof to its settlement layer (L1 or Gateway). See [here](../../settlement_contracts/zkchain_basics.md) for more details.
+
+3. **Updating the MessageRoot on the Settlement Layer**
+   - As the sending chain settles, the Executor facet of settling chain calls the MessageRoot contract, which is then updated with the new L2→L1 logs (see [here](./message_root.md#appending-new-batch-root-leaves)).
```
"As the sending chain settles" - this can be elaborated. What contract and function are called? With what arguments?

nikitastupin-matterlabs replied:
"the Executor facet of settling chain calls the MessageRoot contract" - which function is called?

nikitastupin-matterlabs commented on docs/src/specs/contracts/bridging/interop/overview.md:
```diff
@@ -0,0 +1,50 @@
+3. **Updating the MessageRoot on the Settlement Layer**
+   - As the sending chain settles, the Executor facet of settling chain calls the MessageRoot contract, which is then updated with the new L2→L1 logs (see [here](./message_root.md#appending-new-batch-root-leaves)).
+   - The updated MessageRoot now [commits](https://en.wikipedia.org/wiki/Commitment_scheme) to all L2→L1 logs emitted from the settling chain.
+   - An [event](../../../../../../contracts/l1-contracts/contracts/bridgehub/MessageRoot.sol#L38) is emitted to indicate that the MessageRoot has been updated.
```
The link is broken. Ideally, we’d have event definitions in the doc (including the event name, parameter names and parameter types). Then the doc guides the implementation. But at least the event name is needed so that it’s possible to look it up in the codebase. Just a link may be misleading since the line number may change if we link to a branch or the event may change if we link to a commit.

nikitastupin-matterlabs commented on docs/src/specs/contracts/bridging/interop/overview.md:
```diff
@@ -0,0 +1,50 @@
+
+4. **The Receiving Chain Fetches the Updated MessageRoot**
+   - The receiving chain monitors the Settlement Layer using its EthWatch component.
+   - It detects the event emitted by the MessageRoot contract on the Settlement Layer and saves the updated MessageRoot in its database, corresponding to that block of the Settlement Layer.
```
What happens if two such events are emitted at the same block?

nikitastupin-matterlabs commented on docs/src/specs/contracts/bridging/interop/overview.md:
```diff
@@ -0,0 +1,50 @@
+4. **The Receiving Chain Fetches the Updated MessageRoot**
+   - The receiving chain monitors the Settlement Layer using its EthWatch component.
+   - It detects the event emitted by the MessageRoot contract on the Settlement Layer and saves the updated MessageRoot in its database, corresponding to that block of the Settlement Layer.
+   - **Note:** In some scenarios, alternative hashes might be imported; see [here](./forms_of_finality.md) for additional details.
```
It's not clear what are the alternative hashes even with the link (the word hash occurs 0 times in the linked file).

nikitastupin-matterlabs commented on docs/src/specs/contracts/bridging/interop/overview.md:
```diff
@@ -0,0 +1,50 @@
+   - The receiving chain monitors the Settlement Layer using its EthWatch component.
+   - It detects the event emitted by the MessageRoot contract on the Settlement Layer and saves the updated MessageRoot in its database, corresponding to that block of the Settlement Layer.
+   - **Note:** In some scenarios, alternative hashes might be imported; see [here](./forms_of_finality.md) for additional details.
+   - When creating the next batch, the MessageRoot is imported into the [bootloader](../../../../../../contracts/system-contracts/bootloader/bootloader.yul#L4129). This process saves the MessageRoot in the chain's [L2MessageRootStorage](../../../../../../contracts/system-contracts/contracts/L2MessageRootStorage.sol) contract, against which the Merkle proofs of interop messages will be verified in ...
```
What happens if there are many MessageRoots available for the chain? (e.g. if the sending chain produces batches faster than the receiving chain). Also good to define which function the bootloader calls to store the MessageRoot in the L2MessageRootStorage.

nikitastupin-matterlabs commented on docs/src/specs/contracts/bridging/interop/overview.md:
```diff
@@ -0,0 +1,50 @@
+   - **Note:** In some scenarios, alternative hashes might be imported; see [here](./forms_of_finality.md) for additional details.
+   - When creating the next batch, the MessageRoot is imported into the [bootloader](../../../../../../contracts/system-contracts/bootloader/bootloader.yul#L4129). This process saves the MessageRoot in the chain's [L2MessageRootStorage](../../../../../../contracts/system-contracts/contracts/L2MessageRootStorage.sol) contract, against which the Merkle proofs of interop messages will be verified in ...
+
+5. **Submitting Cross-L2 Transactions on the Destination Chain**
```
Should we use the same term for the receiving chain? Both receiving and destination works but it's good to have consistency across the docs.

nikitastupin-matterlabs commented on docs/src/specs/contracts/bridging/interop/overview.md:
```diff
@@ -0,0 +1,50 @@
+   - The interop transactions (xL2 txs) can now be submitted to the destination chain. The `InteropSwitch/Slingshot` (name to be finalized) component on the sending chain monitors its own InteropCenter contract and detects the emitted `InteropBundleSent` and `InteropTriggerSent` events and corresponding L2→L1 messages.
+   - The InteropSwitch requests the Merkle proofs for the L2→L1 messages. These proofs are the same ones as are used to finalize the L2→L1 messages (see [here](../../settlement_contracts/priority_queue/l1_l2_communication/l2_to_l1.md)).
+   - The event data, combined with the Merkle proofs, is used to construct the interop transactions. This step applies to the standard interop scenario with two bundles and a trigger; in other cases, additional gas payment may be required on the destination chain.
+   - The interop transaction is then submitted to the destination chain.
```
Who submits the transaction? When exactly is it submitted? It reads like the transaction is submitted after the `InteropBundleSent` and `InteropTriggerSent` events are emitted but then the `MessageRoot` is not yet delivered to the receiving chain so the transaction fails.

nikitastupin-matterlabs replied:
Also, is what is the type of the transaction? Is it a new one introduced by the interop?

nikitastupin-matterlabs commented on docs/src/specs/contracts/bridging/interop/overview.md:
```diff
@@ -0,0 +1,50 @@
+
+5. **Submitting Cross-L2 Transactions on the Destination Chain**
+   - The interop transactions (xL2 txs) can now be submitted to the destination chain. The `InteropSwitch/Slingshot` (name to be finalized) component on the sending chain monitors its own InteropCenter contract and detects the emitted `InteropBundleSent` and `InteropTriggerSent` events and corresponding L2→L1 messages.
+   - The InteropSwitch requests the Merkle proofs for the L2→L1 messages. These proofs are the same ones as are used to finalize the L2→L1 messages (see [here](../../settlement_contracts/priority_queue/l1_l2_communication/l2_to_l1.md)).
```
What component provides the proofs in return for the request? What function is called?

nikitastupin-matterlabs commented on docs/src/specs/contracts/bridging/interop/overview.md:
```diff
@@ -0,0 +1,50 @@
+     - Iterates over all calls in the bundle; for each call, it deploys the AliasedAccount contract and specifies the call to be executed. The AliasedAccount then executes the call.
+
+7. **Settling of the Destination Chain**
+   - The destination chain settles on the Settlement Layer (SL). When sealing the batch, the saved `MessageRoot` are sent to the Settlement Layer using an L2→L1 system log, which can be processed during batch execution in the Executor contract.
```
Are we introducing a new system log specifically for saved MessageRoots? If so should we specify it? As far as I remember currently we have ~8 system logs and we rarely add new ones.

nikitastupin-matterlabs replied:
What does `processed` consists of in this case?

Why `can`? What determines if it's processed or not processed?

zkzoomer committed:
rm chain gateway

kelemeno committed:
next attempt

kelemeno committed:
add id number to chain names

kelemeno committed:
Revert "add id number to chain names"

This reverts commit 097e95032600052284b3bd67c8726b8675575023.

kelemeno committed:
try modified internal_id for chains

kelemeno committed:
skip test

kelemeno committed:
print internal id

kelemeno committed:
missing +1

slowli committed:
feat(config): Support serde-style enums in config system (#4055)

## What ❔

- Bumps `smart-config` revision to support serde-style enums (i.e.,
objects with a single field).
- Tests that pre-migration configs are parsed as expected. There's a
single known exception: `da_client.eigen.points_source_url` is still
**NOT** parsed as previously for reasons.

## Why ❔

To make enum config migrations non-mandatory (with a single exception,
see above).

## Is this a breaking change?

- [ ] Yes
- [x] No

## Operational changes

Enum config migrations (i.e., ones for object stores and for the DA
client) are now non-mandatory, with the exception of
`da_client.eigen.points_source_url`.

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] Code has been formatted via `zkstack dev fmt` and `zkstack dev
lint`.

slowli committed:
feat(en): Add revert CLI command for external node (#4053)

## What ❔

Adds a `revert L1_BATCH` command for the EN binary. The command reverts
the node state to the specified batch.

## Why ❔

- This functionality is demanded in practice (e.g., if an EN isn't
updated in time).
- The EN Docker image includes a standalone block_reverter binary, but
using it is inconvenient, given that the way to supply configs to this
binary differs from the EN.

## Is this a breaking change?

- [ ] Yes
- [x] No

## Operational changes

The added command can be used in node maintenance.

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] Code has been formatted via `zkstack dev fmt` and `zkstack dev
lint`.

StanislavBreadless committed:
sync with interop branch

StanislavBreadless committed:
sync with base

zkzoomer committed:
revert ci-core-reusable.yml

zkzoomer committed:
try sleeping

aon committed:
feat: add prividium mode to zkstack explorer (#4079)

## What ❔

Adds prividium mode when deploying a new explorer with zkstack

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- The `Why` has to be clear to non-Matter Labs entities running their
own ZK Chain -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Is this a breaking change?
- [ ] Yes
- [ ] No

## Operational changes
<!-- Any config changes? Any new flags? Any changes to any scripts? -->
<!-- Please add anything that non-Matter Labs entities running their own
ZK Chain may need to know -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev
lint`.

slowli committed:
perf(api): Add observability for `contractAddress` logic (#4066)

## What ❔

Instruments logic that fills `contractAddress` for transaction receipts
on the API server.

## Why ❔

This logic is added recently and could be a reason behind the recent API
server performance degradation.

## Is this a breaking change?

- [ ] Yes
- [x] No

## Operational changes

No operational changes.

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [x] Code has been formatted via `zkstack dev fmt` and `zkstack dev
lint`.

Deniallugo committed:
feat(eth_sender): Add fast finalization into eth_tx_manager (#4070)

## What ❔

Treat Safe block as fast finalized block and mark it correspondigly in
database.

## Why ❔

It's one of the steps for fast finality on L1 

## Is this a breaking change?
- [ ] Yes
- [ ] No

## Operational changes
<!-- Any config changes? Any new flags? Any changes to any scripts? -->
<!-- Please add anything that non-Matter Labs entities running their own
ZK Chain may need to know -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev
lint`.

---------

Signed-off-by: Danil <deniallugo@gmail.com>
Co-authored-by: perekopskiy <53865202+perekopskiy@users.noreply.github.com>

tomg10 committed:
feat(private-rpc): workflow for publishing private-rpc on dockerhub (#4064)

Co-authored-by: Ivan Schasny <31857042+ischasny@users.noreply.github.com>

antonbaliasnikov committed:
ci: add dry publishing in release PR to check for publication issues (#4086)

## What ❔

* [x] add dry publishing in release PR to check for publication issues

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Tests

Tested successfully here by `pull-request` trigger that was removed
after:

https://github.com/matter-labs/zksync-era/actions/runs/15300357238/job/43039139291?pr=4086

## Why ❔

Sometimes we can have issues with `crates.io` publishing due to
dev-dependencies removal with `--locked` publishing or without locked,
we can struggle with minor crates updates that can lead to build issues.

This `--dry-run` in release PR guarantees the publishing success after
the merge of the release PR.

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- The `Why` has to be clear to non-Matter Labs entities running their
own ZK Chain -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Is this a breaking change?
- [ ] Yes
- [x] No

## Operational changes
<!-- Any config changes? Any new flags? Any changes to any scripts? -->
<!-- Please add anything that non-Matter Labs entities running their own
ZK Chain may need to know -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [x] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] Code has been formatted via `zkstack dev fmt` and `zkstack dev
lint`.

Deniallugo committed:
Merge branch 'draft-v29' into sb-precommit-proposal

Deniallugo committed:
Merge branch 'sb-precommit-proposal' into deniallugo-precommit-gateway

StanislavBreadless committed:
use latest contracts

Deniallugo committed:
feat(eth_sender): Fast finalization of eth tx (#4052)

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- The `Why` has to be clear to non-Matter Labs entities running their
own ZK Chain -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

- [ ] Yes
- [ ] No

<!-- Any config changes? Any new flags? Any changes to any scripts? -->
<!-- Please add anything that non-Matter Labs entities running their own
ZK Chain may need to know -->

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev
lint`.

---------

Signed-off-by: Danil <deniallugo@gmail.com>

zkzoomer committed:
Merge remote-tracking branch 'origin/draft-v29' into kl/reduced-interop-support

zkzoomer committed:
merge fixes

Deniallugo committed:
Update contracts hashes

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Fixes for tests

Signed-off-by: Danil <deniallugo@gmail.com>

zkzoomer committed:
rm logs

zkzoomer committed:
Merge remote-tracking branch 'origin/kl/reduced-interop-support' into sma/fix-integration-test-part-three

Deniallugo committed:
Comment tests from vm

Signed-off-by: Danil <deniallugo@gmail.com>

zkzoomer committed:
rm fallback

zkzoomer committed:
Merge pull request #4078 from matter-labs/sma/fix-integration-test-part-three

fix: fix interop integration test

zkzoomer committed:
unskip tests

zkzoomer committed:
split get_chain_log_proof

Deniallugo committed:
Add test for checking precommits

Signed-off-by: Danil <deniallugo@gmail.com>

zkzoomer committed:
use random L2-B url

Deniallugo committed:
Merge branch 'main' into draft-v29

zkzoomer committed:
add CHAINS to .env

Deniallugo committed:
Merge branch 'draft-v29' into deniallugo-precommit-gateway

zkzoomer committed:
set l2-b to have no custom token

zkzoomer committed:
minor changes

zkzoomer committed:
Merge remote-tracking branch 'origin/draft-v29' into kl/reduced-interop-support

zkzoomer committed:
merge fixes

StanislavBreadless committed:
use latest contracts

zkzoomer committed:
Merge remote-tracking branch 'origin/main' into draft-v29

zkzoomer committed:
merge fixes

zkzoomer committed:
Merge remote-tracking branch 'origin/draft-v29' into kl/reduced-interop-support

zkzoomer committed:
set CHAINS on integration tests

zkzoomer committed:
set CHAINS on run_on_all_chains.sh

StanislavBreadless committed:
sync with base

StanislavBreadless committed:
fmt

StanislavBreadless committed:
lint fix

zkzoomer committed:
revert to era-validium

zkzoomer committed:
use no empty_leaf_hash

zkzoomer committed:
use no empty_leaf_hash

StanislavBreadless committed:
fix unit tests

StanislavBreadless committed:
fix zkstack in upgrades

StanislavBreadless committed:
Merge branch 'kl/reduced-interop-support' into sb-precommit-proposal-addresses-in-vt-v29

zkzoomer committed:
use default impl

zkzoomer committed:
only load MAX_MSG_ROOTS_IN_BATCH

zkzoomer committed:
missing dependency

zkzoomer committed:
version renaming

zkzoomer committed:
linting

StanislavBreadless committed:
use latest contracts

StanislavBreadless committed:
use latest contracts

zkzoomer committed:
try sorting

zkzoomer committed:
comment fixes

StanislavBreadless committed:
sync with base

StanislavBreadless committed:
sync with base

StanislavBreadless committed:
fix zkstack cli changes

StanislavBreadless committed:
sync with 4009

StanislavBreadless committed:
fix lint

StanislavBreadless committed:
upd contracts

Deniallugo committed:
Fix zksatck

Signed-off-by: Danil <deniallugo@gmail.com>

zkzoomer committed:
minor fixes

Deniallugo committed:
Adapt block reverter for new interface

Signed-off-by: Danil <deniallugo@gmail.com>

zkzoomer committed:
use single zksync-ethers

zkzoomer committed:
use single zksync-ethers

Deniallugo committed:
Merge branch 'kl/reduced-interop-support' into sb-precommit-proposal-addresses-in-vt-v29

zkzoomer committed:
Revert "use single zksync-ethers"

This reverts commit 82271bd8c8f9fe4711b90d7f3fea578f86b1400c.

Deniallugo committed:
Merge branch 'kl/reduced-interop-support' into sb-precommit-proposal-addresses-in-vt-v29

Deniallugo committed:
Move the logic about setting eth_txs out of dal

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Small refactoring

Signed-off-by: Danil <deniallugo@gmail.com>

zkzoomer committed:
read gw chain id

zkzoomer committed:
bump max count

zkzoomer committed:
Merge pull request #4122 from matter-labs/sma/single-zksync-ethers

fix: single zksync-ethers

zkzoomer committed:
bump zksync-ethers

Deniallugo committed:
change transaction.l1_batch to miniblock.l1_batch

Signed-off-by: Danil <deniallugo@gmail.com>

zkzoomer committed:
order interop_roots from database

zkzoomer committed:
flip getL2bUrl

zkzoomer committed:
fetch one

zkzoomer committed:
move and reformat get_last_sent_successfully_eth_tx_id_by_batch_and_op

zkzoomer committed:
rename to sl_block_number

zkzoomer committed:
comments

zkzoomer committed:
renaming

zkzoomer committed:
typos

zkzoomer committed:
polish interop scripts

zkzoomer committed:
cleanup

zkzoomer committed:
bump contracts

zkzoomer committed:
update genesis

Deniallugo committed:
Update some queries and index

Signed-off-by: Danil <deniallugo@gmail.com>

zkzoomer committed:
rename to InteropMode

zkzoomer committed:
Merge pull request #4113 from matter-labs/sma/fix-consensus-integration-test

fix: fix consensus integration test

zkzoomer committed:
Merge remote-tracking branch 'origin/kl/reduced-interop-support' into sb-precommit-proposal-addresses-in-vt-v29

zkzoomer committed:
Merge remote-tracking branch 'origin/kl/reduced-interop-support' into sma/reduced-interop-support-cleanup

zkzoomer committed:
bump zksync-ethers interop support

Deniallugo committed:
Fix race condition

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Use 0 batch as default one

Signed-off-by: Danil <deniallugo@gmail.com>

zkzoomer committed:
bump zksync-ethers interop support

kelemeno committed:
Merge branch 'kl/reduced-interop-support' of github.com:matter-labs/zksync-era into kl/interop-support

Deniallugo committed:
Check that the batch we want to commit belongs to the one we want

Signed-off-by: Danil <deniallugo@gmail.com>

kelemeno committed:
small fixes for working condition

Deniallugo committed:
Support for disabling precommits

Signed-off-by: Danil <deniallugo@gmail.com>

zkzoomer committed:
Merge pull request #4130 from matter-labs/sma/reduced-interop-support-cleanup

chore: reduced interop support cleanup

zkzoomer committed:
Merge remote-tracking branch 'origin/kl/reduced-interop-support' into sb-precommit-proposal-addresses-in-vt-v29

zkzoomer committed:
try increase timeout

zkzoomer committed:
Merge pull request #3759 from matter-labs/kl/reduced-interop-support

feat: reduced interop support

zkzoomer committed:
Merge remote-tracking branch 'origin/main' into draft-v29

zkzoomer committed:
fix genesis

zkzoomer committed:
revert contracts

zkzoomer committed:
regenerate genesis

zkzoomer committed:
Merge branch 'main' into draft-v29

itegulov committed:
feat(api): implement `unstable_gasPerPubdata` (#4124)

Adds `unstable_gasPerPubdata` that is to be stabilized as
`zks_gasPerPubdata` in the future.

Preparation for deprecation of `zks_estimateFee`

- [ ] Yes
- [x] No

Whitelist  `unstable_gasPerPubdata` in proxyd

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [x] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [x] Code has been formatted via `zkstack dev fmt` and `zkstack dev
lint`.

zkzoomer committed:
temp ignore consensus registry tests

zkzoomer committed:
minor fixes

zkzoomer committed:
minor fixes cont
oopsies

zkzoomer committed:
rm unused

zkzoomer committed:
test fixes

zkzoomer committed:
fix test

zkzoomer committed:
Merge remote-tracking branch 'origin/draft-v29' into sb-precommit-proposal-addresses-in-vt-v29

zkzoomer committed:
merge fixes

zkzoomer committed:
rm logs

zkzoomer committed:
rm unused

zkzoomer committed:
add new fast vm version

zkzoomer committed:
add interop execute gas cost contribution

zkzoomer committed:
add l1_message_root_address

zkzoomer committed:
Revert "rm logs"

This reverts commit 00998e921a9fc9c285d54d7f03b6263d8f6434b1.

zkzoomer committed:
Merge pull request #4170 from matter-labs/sma/lagging-reduced-interop-todos

fix: lagging interop todos

zkzoomer committed:
Merge branch 'main' into draft-v29

zkzoomer committed:
Merge remote-tracking branch 'origin/draft-v29' into sb-precommit-proposal-addresses-in-vt-v29

zkzoomer committed:
bump contracts

zkzoomer committed:
update vm constants

zkzoomer committed:
rm unused

zkzoomer committed:
update genesis

zkzoomer committed:
minor fixes

Deniallugo committed:
Make fast finality faster

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Use confirmations only for gateway

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Do not wait for finalization

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Try to poll more frequent

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Load the sl batch number from log

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Load the sl batch number from log

Signed-off-by: Danil <deniallugo@gmail.com>

zkzoomer committed:
Merge remote-tracking branch 'origin/main' into draft-v29

Deniallugo committed:
feat(en): Support interop roots inside en

Signed-off-by: Danil <deniallugo@gmail.com>

zkzoomer committed:
use zksync-ethers release

zkzoomer committed:
review fixes

zkzoomer committed:
related fixes

zkzoomer committed:
linting

zkzoomer committed:
get max interop roots from state keeper

zkzoomer committed:
minor fixes

zkzoomer committed:
revert "minor fixes"

This reverts commit ac02a079d1937958c9f93b96f80218c9b2fffe63.

zkzoomer committed:
rm old dead code flags

zkzoomer committed:
derive Copy

zkzoomer committed:
rm unused

zkzoomer committed:
rm unused

zkzoomer committed:
rm confirmed_at_block

Deniallugo committed:
make interop root prettier

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
some debug

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Upsert the interop roots

Signed-off-by: Danil <deniallugo@gmail.com>

zkzoomer committed:
fix tests

zkzoomer committed:
rm old zksync-ethers

zkzoomer committed:
Merge remote-tracking branch 'origin/draft-v29' into sb-precommit-proposal-addresses-in-vt-v29

zkzoomer committed:
unifying to is_pre_interop_fast_blocks

zkzoomer committed:
linting

zkzoomer committed:
fix tests

zkzoomer committed:
Merge remote-tracking branch 'origin/main' into draft-v29

zkzoomer committed:
rm get_last_sent_and_confirmed_eth_storage_tx

Deniallugo committed:
Speed up the sync

Signed-off-by: Danil <deniallugo@gmail.com>

zkzoomer committed:
Merge branch 'draft-v29' of https://github.com/matter-labs/zksync-era into draft-v29

Deniallugo committed:
Fix limit

Signed-off-by: Danil <deniallugo@gmail.com>

zkzoomer committed:
unify migration

zkzoomer committed:
use StorageInteropRoots

zkzoomer committed:
define EncodingVersion

zkzoomer committed:
cleanup

zkzoomer committed:
rename to chain_log_proof_until_msg_root

Deniallugo committed:
Fix ordering

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Merge branch 'draft-v29' into deniallugo-support-interop-roots-en

zkzoomer committed:
change to strict equality

zkzoomer committed:
rm batch_chain_global_merkle_path

zkzoomer committed:
renaming for clarity

zkzoomer committed:
Merge remote-tracking branch 'origin/draft-v29' into sb-precommit-proposal-addresses-in-vt-v29

zkzoomer committed:
more renaming

zkzoomer committed:
Merge remote-tracking branch 'origin/main' into draft-v29

kelemeno committed:
small script fix

Deniallugo committed:
debug

Signed-off-by: Danil <deniallugo@gmail.com>

zkzoomer committed:
fix: add InteropBatchRootProcessor (#4223)

## What ❔

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- The `Why` has to be clear to non-Matter Labs entities running their
own ZK Chain -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Is this a breaking change?
- [ ] Yes
- [ ] No

## Operational changes
<!-- Any config changes? Any new flags? Any changes to any scripts? -->
<!-- Please add anything that non-Matter Labs entities running their own
ZK Chain may need to know -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev
lint`.

zkzoomer committed:
Merge remote-tracking branch 'origin/main' into draft-v29

Deniallugo committed:
Remove redundant db call

Signed-off-by: Danil <deniallugo@gmail.com>

zkzoomer committed:
refer to main docs/

zkzoomer committed:
Merge remote-tracking branch 'origin/main' into draft-v29

zkzoomer committed:
rm added docs, using kl/interop-docs as source of truth

Deniallugo committed:
clean interop roots

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Merge branch 'draft-v29' into deniallugo-support-interop-roots-en

zkzoomer committed:
Merge remote-tracking branch 'origin/draft-v29' into sb-precommit-proposal-addresses-in-vt-v29

Deniallugo committed:
fix rollback l2 block

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
clean pr

Signed-off-by: Danil <deniallugo@gmail.com>

zkzoomer committed:
Merge pull request #4188 from matter-labs/deniallugo-support-interop-roots-en

feat(en): Support interop roots inside en

Deniallugo committed:
return correct statuses back

Signed-off-by: Danil <deniallugo@gmail.com>

Deniallugo committed:
Close batches for gateway slower

Signed-off-by: Danil <deniallugo@gmail.com>

zkzoomer committed:
Revert "try increase timeout"

This reverts commit 95bc6e8374d43443e896a18987615bdbe04b9800.

zkzoomer committed:
Merge remote-tracking branch 'origin/sb-precommit-proposal-addresses-in-vt-v29' into sma/change-memory-layout

zkzoomer committed:
rm private rpc workflow, as its not used on main

zkzoomer committed:
try fetch full history

zkzoomer committed:
Merge remote-tracking branch 'origin/main' into draft-v29

zkzoomer committed:
rm commented code

zkzoomer committed:
rename to get_number_of_applied_interop_roots

zkzoomer committed:
rm interop_roots_assigned

zkzoomer committed:
support encoding versions in consistency checker

zkzoomer committed:
rename to InteropSupported

zkzoomer committed:
use L2BlockApplicationConfig

zkzoomer committed:
spacing

zkzoomer committed:
fix get_preexisting_blocks_number

zkzoomer committed:
rm interop_roots from StoredL2BlockEnv

perekopskiy committed:
feat: update is_pre_fast_blocks (#4249)

## What ❔

updates is_pre_fast_blocks and fixes some unit tests

## Why ❔

enable sub-second blocks for v29

## Is this a breaking change?
- [ ] Yes
- [ ] No

## Operational changes
<!-- Any config changes? Any new flags? Any changes to any scripts? -->
<!-- Please add anything that non-Matter Labs entities running their own
ZK Chain may need to know -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev
lint`.

zkzoomer committed:
Merge remote-tracking branch 'origin/main' into draft-v29

zkzoomer committed:
fix: various draft-v29 fixes (#4254)

## What ❔

Includes:
- [x] Removal of `BatchOrBlockNumber`
- [x] Removal of `received_timestamp`
- [x] Recent major VM changes from
https://github.com/matter-labs/zksync-era/pull/4221

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- The `Why` has to be clear to non-Matter Labs entities running their
own ZK Chain -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Is this a breaking change?
- [ ] Yes
- [ ] No

## Operational changes
<!-- Any config changes? Any new flags? Any changes to any scripts? -->
<!-- Please add anything that non-Matter Labs entities running their own
ZK Chain may need to know -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev
lint`.

zkzoomer committed:
Merge remote-tracking branch 'origin/main' into draft-v29

zkzoomer committed:
merge fixes

zkzoomer committed:
Merge remote-tracking branch 'origin/draft-v29' into sb-precommit-proposal-addresses-in-vt-v29

zkzoomer committed:
fix: merge processors (#4256)

## What ❔

Merges `BatchRootProcessor` and `BatchRootProcessorInterop` into a
single one

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- The `Why` has to be clear to non-Matter Labs entities running their
own ZK Chain -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Is this a breaking change?
- [ ] Yes
- [ ] No

## Operational changes
<!-- Any config changes? Any new flags? Any changes to any scripts? -->
<!-- Please add anything that non-Matter Labs entities running their own
ZK Chain may need to know -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev
lint`.

zkzoomer committed:
Merge remote-tracking branch 'origin/sb-precommit-proposal-addresses-in-vt-v29' into sma/change-memory-layout

zkzoomer committed:
fix: change memory layout II (#4239)

## What ❔

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- The `Why` has to be clear to non-Matter Labs entities running their
own ZK Chain -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Is this a breaking change?
- [ ] Yes
- [ ] No

## Operational changes
<!-- Any config changes? Any new flags? Any changes to any scripts? -->
<!-- Please add anything that non-Matter Labs entities running their own
ZK Chain may need to know -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev
lint`.

zkzoomer committed:
Merge remote-tracking branch 'origin/main' into draft-v29

zkzoomer committed:
Merge remote-tracking branch 'origin/draft-v29' into sb-precommit-proposal-addresses-in-vt-v29

kelemeno committed:
test bwg

kelemeno committed:
changed files to run e2e

StanislavBreadless committed:
cherry pick commit

StanislavBreadless committed:
fmt

kelemeno committed:
test bwg (#4278)

## What ❔

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- The `Why` has to be clear to non-Matter Labs entities running their
own ZK Chain -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Is this a breaking change?
- [ ] Yes
- [ ] No

## Operational changes
<!-- Any config changes? Any new flags? Any changes to any scripts? -->
<!-- Please add anything that non-Matter Labs entities running their own
ZK Chain may need to know -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev
lint`.

---------

Co-authored-by: Stanislav Breadless <stanislavbezkor@gmail.com>

kelemeno committed:
returning applied interop roots num

kelemeno committed:
linting

kelemeno committed:
apply bootloader memory prover fixes

kelemeno committed:
linting

kelemeno committed:
Merge pull request #4279 from matter-labs/kl/test-bwg-2

Kl/test bwg 2

kelemeno committed:
Merge branch 'sb-precommit-proposal-addresses-in-vt-v29' of github.com:matter-labs/zksync-era into sma/change-memory-layout

kelemeno committed:
genesis

kelemeno committed:
Merge branch 'kl/bump-fully' of github.com:matter-labs/zksync-era into kl/interop-support

kelemeno committed:
print genesis

kelemeno committed:
more genesis

kelemeno committed:
linting and genesis

kelemeno committed:
ci generate genesis

deniallugo-ml commented:
Closing this PR as it has been open without activity for over a year and the codebase has moved on significantly since it was opened. The changes would need a substantial rebase at minimum, and the original context may no longer apply.

If this addresses something that is still relevant, please open a fresh PR against the current `main` with an updated diff — we are happy to review it.

Thank you for the contribution attempt.
